### PR TITLE
Ranged Attack Speed Upgrade

### PR DESCRIPTION
**Pull request by kattalainen**
Attack speed upgrade, kinda self explanatory. As usual you can comment out other upgrades in UpgradeManager.cpp from line 145 to 220 to only get this upgrade in the shop.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeRangeAttackSpeed.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	m_UpgradeFactor = m_AttackPerSecond * (static_cast<float>(m_Percent)/100.0);
+}
+
+UpgradeRangeAttackSpeed::~UpgradeRangeAttackSpeed()
+{
+}
+
+void UpgradeRangeAttackSpeed::ApplyStat()
+{
+	m_AttackPerSecond += m_UpgradeFactor;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeRangeAttackSpeed.cpp]: Should be based directly on level, i.e. (1.0f / m_BaseAttackSpeed) + (m_UpgradeFactor * (m_Level + 1)).
Same edit should also be made in MeleeAttackSpeed

File: StortSpelProjekt/Game/src/main.cpp [hunk 2]
```diff
@@ -227,7 +227,7 @@ Scene* GameScene(SceneManager* sm)
 
     melc->SetDamage(10);
     melc->SetAttackInterval(1.0);
-    ranc->SetAttackInterval(0.8);
+    ranc->SetAttackInterval(1.0);
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/main.cpp:230]: This change should be done also in Player's onResetGame function

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
I like the icons. Apart from the changes requested, the shop needs three small fixes. This is not your task, but if you want to fix it in this pull request contact me and I will tell you what needs to be done

**Review by jocke1995 — DISMISSED** [review]
gj

**Review by jocke1995 — DISMISSED** [review]
yes

**Review by LeoWikstrom — APPROVED** [review]

**Review by jocke1995 — APPROVED** [review]
wow! Good job :) amazing!

**Pull request merged**